src/context/filterContext/filterContext.jsx

```diff
@@ -7,20 +7,21 @@ import { sortDataByRange } from '../../reducerFunctions/filterReducer/reducerUti
 import { filterByRating } from '../../reducerFunctions/filterReducer/reducerUtilities/filterRatingFunc';
 import { filterByPrice } from '../../reducerFunctions/filterReducer/reducerUtilities/filterPriceFunc';
 import { filterDataByCategory } from '../../reducerFunctions/filterReducer/reducerUtilities/filterCategoryFunc';
-
+import { filterBySearch } from "../../reducerFunctions/filterReducer/reducerUtilities/filterSearch";
 
 
 const FilterContext = createContext(null);
 
 const FilterProvider = ({ children }) => {
     const { productList } = useProduct();
-    const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
+    const [state, dispatch] = useReducer(filterReducer, { sortbyrange: "", sortBySearch: "", sortByRating: "", priceValue: 160000, category: { isBike: false, isScooter: false } });
 
     const sortedData = sortDataByRange(productList, state.sortbyrange);
     const filteredData = filterByRating(sortedData, state.sortByRating);
     const filterDataByPrice = filterByPrice(filteredData, state.priceValue);
     const dataByCategory = filterDataByCategory(filterDataByPrice, state.category.isBike, state.category.isScooter);
```

Comment:
this is also correct , just a suggestion if you have time you can try filter chaining or compose

Reply:
thank you, i will do change that